pkg/route/validation/validation_test.go

```diff
@@ -843,6 +981,522 @@ func TestValidateRoute(t *testing.T) {
 	}
 }
 
+// This unit test tests if the Actions to set/delete headers were passed correctly in the Route Spec and checks if an appropriate error message was provided.
+func TestValidateHeaders(t *testing.T) {
+	var headerNameXFrame = "X-Frame-Options"
+	var headerNameXSS = "X-XSS-Protection"
+	var headerNameHSTS = "Strict-Transport-Security"
+	var headerNameProxy = "Proxy"
+	var headerNameCookie = "Cookie"
+	var headerNameSetCookie = "Set-Cookie"
+	tests := []struct {
+		name                 string
+		route                *routev1.Route
+		expectedErrorMessage string
+		headerType           string
+	}{
+		{
+			name: "Should give an error on attempt to delete the HSTS header.",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "wildcardpolicy",
+					Namespace: "foo",
+				},
+				Spec: routev1.RouteSpec{
+					Host:           "subdomain.wildcard.test",
+					To:             createRouteSpecTo("serviceName", "Service"),
+					WildcardPolicy: routev1.WildcardPolicySubdomain,
+					HTTPHeaders: &routev1.RouteHTTPHeaders{
+
+						Actions: routev1.RouteHTTPHeaderActions{
+							Response: []routev1.RouteHTTPHeader{
+								{
+
+									Name: headerNameHSTS,
+									Action: routev1.RouteHTTPHeaderActionUnion{
+										Type: routev1.Delete,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedErrorMessage: "spec.httpHeaders.actions.response[0].name: Unsupported value: \"Strict-Transport-Security\": supported values: \"All headers except strict-transport-security,cookie,set-cookie\"",
```

Comment:
The error should mention proxy too, right?  I suggest adding a string that captures all the forbidden headers, and using that in all the error formatting that requires it.

Reply:
I wonder how this passes?

Reply:
fixed it

Reply:
https://github.com/openshift/library-go/commit/398bba91d32aa80169a697d8b06ba0ff5edc5774#diff-51cbcb9e560e512d7dba0cd93dab386055814a27e2c7903ad93bce92c58fd18bR1025